### PR TITLE
Add verifiedbeaconreport and pendingbeaconreport

### PR DESCRIPTION
**Pull request by jamescowens**
This adds beaconconvergence and pendingbeaconreport to output verifed and pending beacons.

### CODE REVIEW

File: src/scraper/scraper.cpp [hunk 1]
```diff
@@ -4925,6 +4925,41 @@ ScraperStatsAndVerifiedBeacons GetScraperStatsAndVerifiedBeacons(const Converged
     return stats_and_verified_beacons;
 }
 
+// This is for rpc report functions.
+ScraperPendingBeaconMap GetPendingBeaconsForReport()
+{
+    return GetConsensusBeaconList().mPendingMap;
```
Comment (cyrossignol) [inline src/scraper/scraper.cpp:4931]: If we use the consensus list, none of the RPCs will show pending beacons advertised within the last hour. Those are available from `NN::GetBeaconRegistry().PendingBeacons()` directly. Maybe we need `beaconreport`, `pendingbeaconreport`, and `beaconconvergence` functions, where the last provides introspection into the scraper's data structures. What do you think?

I know I said that I find separate RPCs more useful, but it seems to make more sense to group the last one together after thinking about it in this way.

File: src/rpcblockchain.cpp [hunk 2]
```diff
@@ -759,6 +774,80 @@ UniValue beaconreport(const UniValue& params, bool fHelp)
...
+
+        verified_from_global.push_back(entry);
+    }
+
+    results.pushKV("verified beacons from scraper global", verified_from_global);
+
+
+    UniValue verified_from_convergence(UniValue::VARR);
+
+    const ScraperPendingBeaconMap& verified_beacons_from_convergence = GetVerifiedBeaconsForReport(false);
```
Comment (cyrossignol) [inline src/rpcblockchain.cpp]: Did you mean to declare the `verified_beacons_from_convergence` variable as a reference? I'm not sure if C++ will always let the temporary object live until the function finishes using it.

Comment (jamescowens) [reply src/rpcblockchain.cpp]: Nope. I should fix that.

File: src/scraper/scraper.cpp (hunk 1 above)
Comment (jamescowens) [reply src/scraper/scraper.cpp:4931]: So what I think you are saying is taking the current method of the pendingbeaconreport, and put it in the verifiedbeaconreport and rename that beaconconvergence, then replace the guts of the pendingbeaconreport with a direct access to .PendingBeacons. The only problem with that approach is I don't think the .PendingBeacons has the max depth cutoff, but maybe that is really not an issue. It definitely won't have the ladder effect at the top, which is what you want.

Comment (cyrossignol) [reply src/scraper/scraper.cpp:4931]: Yeah, there's no max-depth cutoff for `BeaconRegistry::Beacons()` or `BeaconRegistry::PendingBeacons()`. This matches the old `AppCache` for which the RPCs just displayed everything in memory. Useful for debugging to see how state mutates or lingers, but we can add a check for `Expired()` as in `GetConsensusBeaconList()` if you think that makes sense.

Comment (jamescowens) [reply src/scraper/scraper.cpp:4931]: I think that isn't necessary.

**Review by cyrossignol — APPROVED** [review]
Looks good 👍

**Pull request merged**